rerun_py/rerun_sdk/rerun/event.py

```diff
@@ -0,0 +1,164 @@
+    @property
+    def type(self) -> Literal["pause"]:
+        return "pause"
+
+
+@dataclass
+class TimeUpdateEvent(ViewerEventBase):
+    @property
+    def type(self) -> Literal["time_update"]:
+        return "time_update"
+
+    time: float
+
+
+@dataclass
+class TimelineChangeEvent(ViewerEventBase):
+    @property
+    def type(self) -> Literal["timeline_change"]:
+        return "timeline_change"
+
+    timeline: str
+    time: float
+
+
+@dataclass
+class SelectionChangeEvent(ViewerEventBase):
+    @property
+    def type(self) -> Literal["selection_change"]:
+        return "selection_change"
+
+    items: list[SelectionItem]
+
+
+# Union type for all possible event types
+ViewerEvent = Union[PlayEvent, PauseEvent, TimeUpdateEvent, TimelineChangeEvent, SelectionChangeEvent]
+
+
+def _viewer_event_from_json_str(json_str: str) -> ViewerEvent:
+    data = json.loads(json_str)
+
+    event_type: str = data["type"]
+    app_id: str = data["application_id"]
+    recording_id: str = data["recording_id"]
+
+    if event_type == "play":
+        return PlayEvent(application_id=app_id, recording_id=recording_id)
+
+    elif event_type == "pause":
+        return PauseEvent(application_id=app_id, recording_id=recording_id)
+
+    elif event_type == "time_update":
+        return TimeUpdateEvent(
+            application_id=app_id,
+            recording_id=recording_id,
+            time=data["time"],
+        )
+
+    elif event_type == "timeline_change":
+        return TimelineChangeEvent(
+            application_id=app_id,
+            recording_id=recording_id,
+            timeline=data["timeline"],
+            time=data["time"],
+        )
+
+    elif event_type == "selection_change":
+        items: list[SelectionItem] = []
+        for item in data["items"]:
+            if item["type"] == "entity":
+                items.append(
+                    EntitySelectionItem(
+                        entity_path=item["entity_path"],
+                        instance_id=item.get("instance_id", None),
+                        view_name=item.get("view_name", None),
+                        position=item.get("position", None),
+                    )
+                )
+            elif item["type"] == "view":
+                items.append(
+                    ViewSelectionItem(
+                        view_id=item["view_id"],
+                        view_name=item["view_name"],
+                    )
+                )
+            elif item["type"] == "container":
+                items.append(
+                    ContainerSelectionItem(
+                        container_id=item["container_id"],
+                        container_name=item["container_name"],
+                    )
+                )
+
+        return SelectionChangeEvent(
+            application_id=app_id,
+            recording_id=recording_id,
+            items=items,
+        )
+
+    else:
+        raise ValueError(f"Unknown event type: {event_type}")
```

Comment:
Pro-tip: always add quotes, or you'll tear your hair out trying to figure out why the string `'container '` doesn't match

```suggestion
        raise ValueError(f"Unknown event type: '{event_type}'")
```